modules/light-clients/07-tendermint/client_state_test.go

```diff
@@ -71,6 +71,71 @@ func (suite *TendermintTestSuite) TestStatus() {
 	}
 }
 
+func (suite *TendermintTestSuite) TestGetTimestampAtHeight() {
+	var (
+		path   *ibctesting.Path
+		height exported.Height
+	)
+	expectedTimestamp := time.Unix(1, 0)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expErr   error
+	}{
+		{
+			"success",
+			func() {},
+			nil,
+		},
+		{
+			"failure: consensus state not found for height",
+			func() {
+				clientState := path.EndpointA.GetClientState()
+				height = clientState.GetLatestHeight().Increment()
+			},
+			clienttypes.ErrConsensusStateNotFound,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.SetupClients(path)
+
+			clientState := path.EndpointA.GetClientState()
+			height = clientState.GetLatestHeight()
+
+			store := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), path.EndpointA.ClientID)
+
+			// grab consensusState from store and update with a predefined timestamp
+			consensusState := path.EndpointA.GetConsensusState(height)
+			tmConsensusState, ok := consensusState.(*ibctm.ConsensusState)
+			suite.Require().True(ok)
+
+			tmConsensusState.Timestamp = expectedTimestamp
+			path.EndpointA.SetConsensusState(tmConsensusState, height)
```

Comment:
Just a nit, but I feel like you could just leave this line in the loop over the test cases:

```suggestion
			store := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), path.EndpointA.ClientID)
```

And then have this in the `malleate` function for the success case.

```
clientState := path.EndpointA.GetClientState()
height = clientState.GetLatestHeight()

// grab consensusState from store and update with a predefined timestamp
consensusState := path.EndpointA.GetConsensusState(height)
tmConsensusState, ok := consensusState.(*ibctm.ConsensusState)
suite.Require().True(ok)

tmConsensusState.Timestamp = expectedTimestamp
path.EndpointA.SetConsensusState(tmConsensusState, height)
```

Reply:
it's nice having success be a no-op, otherwise you need to likely include the same code in failure test cases (to ensure it'd succeed if the failure logic was missing)

The expected timestamp could be grabbed from the existing consensus state, rather than overwriting it, but this is fine by me